android/modules/app/src/java/ti/modules/titanium/app/AppModule.java

```diff
@@ -213,14 +217,23 @@ public boolean getAccessibilityEnabled()
 	@Kroll.method(name = "_restart")
 	public void restart()
 	{
-		Application app = (Application) KrollRuntime.getInstance().getKrollApplication();
-		Intent i = app.getPackageManager().getLaunchIntentForPackage(app.getPackageName());
-		i.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
-		i.addFlags(Intent.FLAG_ACTIVITY_RESET_TASK_IF_NEEDED);
-		i.addCategory(Intent.CATEGORY_LAUNCHER);
-		i.setAction(Intent.ACTION_MAIN);
+		KrollRuntime runtime = KrollRuntime.getInstance();
+
+		// prevent termination of root activity via TiBaseActivity.shouldFinishRootActivity()
+		TiBaseActivity.canFinishRoot = false;
```

Comment:
Once we set `canFinishRoot` to `false`, then backing out of the first Ti.UI.Window in a restarted window will now show the launcher activity, right? We may need to reset `canFinishRoot` back to `true` after the restart.

Perhaps this shouldn't be an exposed public global and handle it via a method.

Reply:
`canFinishRoot` returns to true on [Line 229](https://github.com/appcelerator/titanium_mobile/pull/9946/files#diff-6605aa182d97e7cfaab1dd4f36bcf791R229)